pkg/resources/server_test.go

```diff
@@ -297,14 +323,29 @@ var _ = Describe("resourceServer tests", func() {
 			deprecatedSockDir = deprecatedSockDirBackup
 		})
 		BeforeEach(func() {
-			fs = utils.FakeFilesystem{}
+			fs = utils.FakeFilesystem{
+				Files: map[string][]byte{fakeSocketName: []byte("")},
+			}
 			cleanTemp = fs.Use()
 			deprecatedSockDir = fs.RootDir
+			fakeSocketPath = path.Join(fs.RootDir, fakeSocketName)
 		})
 		AfterEach(func() {
 			cleanTemp()
 		})
-		It("Watch socket and send notification of create successfully", func() {
+		It("Watch socket then stop watcher", func() {
```

Comment:
is there a test case that watches the socket, then delete the socket  so the `lstat` portion kicks in ?